flaml/automl/model.py

```diff
@@ -2752,8 +2751,6 @@ def __init__(self, start_time, deadline, free_mem_ratio):
     def check_resource_limits(self, current_time, current_iteration, mllib):
         if (mllib == "xgb" and current_iteration == 0) or (mllib == "cat" and current_iteration == 1):
             self._time_per_iter = current_time - self.start_time
-        if current_time + self._time_per_iter > self.deadline:
-            return False

```

Comment:
This is needed.

Reply:
Hi @thinkall,
I added the `retrain_full = True` to my own test case, and it didn't seem to affect my results. I looked at the documentation, which describes `retrain_full` as follows:
'_default=True | whether to retrain the selected model on the full training data when using holdout. True - retrain only after search finishes; False - no retraining; 'budget' - do best effort to retrain without violating the time budget._'
My tests use cross-validation, so it makes sense that `retrain_full` would have no effect. 

I re-added the if statement you highlighted above, and  attempted to see if I could find a FLAML configuration which returns a model which I am unable to replicate the result of. Unfortunately, I did find such a configuration. I've made this case into its own unit test on `test_regression.py` - it's called `test_reproducibility_of_catboost_regression_model`.
I'm not sure how reproducible this error will be when run on different hardware, so I'll paste one of the failing results I get below:
![image](https://github.com/user-attachments/assets/7ee8320a-3776-4288-bbfe-0fb4ca14f1ec)

It is still my opinion that the above if statement should be removed - at least for the CatBoost case (I haven't tested XGBoost in depth yet). The runtime reduction from finishing Catboost training early is not, in my opinion, worth the trade-off of risking returning the wrong model to the user.

Please let me know your thoughts!